blackjax/mcmc/hmc.py

```diff
@@ -41,6 +41,18 @@ class HMCState(NamedTuple):
     logdensity_grad: ArrayTree
 
 
+class DynamicHMCState(NamedTuple):
+    """State of the dynamic HMC algorithm.
+
+    Adds a utility array for generating a pseudo or quasi-random sequence of
+    number of integration steps.
+
+    """
+
+    hmc_state: HMCState
```

Comment:
On second thought, let's unpack this and do
```python
class DynamicHMCState(NamedTuple):
    """State of the HMC algorithm.
    The HMC algorithm takes one position of the chain and returns another
    position. In order to make computations more efficient, we also store
    the current logdensity as well as the current gradient of the logdensity.
    """
    position: ArrayTree
    logdensity: float
    logdensity_grad: ArrayTree
    random_generator_arg: Array
```
The reason being that we dont need to do additional unpack and the output of static and dynamic hmc are mostly the same.